### PR TITLE
Add release action

### PR DESCRIPTION
**Pull request by McGiverGim**
This is for discussion, I'm not too sure if this can be useful.

This creates a new Github Action workflow, executed manually, that prepares a draft with the release (final or RC) of the Configurator.

It builds all the artifacts for Linux, Windows, MacOS and Android, and creates a draft release that we can "modify" before publishing. In this way we don't need to have all the systems and build manually to create the release.

It asks for the release name/tag name/default release notes.

image

After finished, it creates something like this:

image

Where we can modify and press publish when we want. It will create the tag and the release.

Some things I have observed:
- For Android, it builds a debug version and not a release version for two reasons, that we will need to look into if we want an automated release build. 
1. We have a "custom" step in the build system for the Android build when is a release version where it prompts the user for the version that it must use, and of course it fails when is an automated build. We can discuss if it is necessary or we can get it from the `package.json` always:
https://github.com/betaflight/betaflight-configurator/blob/51a6b60b0a002941a96fc5fe71f170180cd8cc54/gulpfile.js#L298-L316
2. We will need to have in a GitHub SECRET or somewhere the correct certificate to use to generate the Android Store build. I don't know who has it, I suppose that @mikeller or maybe someone more. So we need to "remove" the debug Android build and upload the correct one in the draft.
- The version is taken from the `package.json`, to create the RC versions I don't know how we are doing it now. @blckmn can you explain? I suppose modify this file in local only and create versions? I think remember there was a problem with this, so we will need to test it and modify the `package.json` before generating the version. 

I wait for your thoughts... The Configurator is the most complicated, do the same for the Firmware or the Blackbox will be easier.

### CODE REVIEW

**Comment by blckmn** [comment]
I have yet to look at this in total... but I reckon a release build "on tag creation" would be great. Such that if you "create" the release in draft in this repo (by adding the tag), then the build is kicked off, and the artifacts are then uploaded to that release (matched by tag).

**Comment by haslinghuis** [comment]
Already prepared the version update after release: https://github.com/betaflight/betaflight-configurator/pull/2835

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
> I have yet to look at this in total... but I reckon a release build "on tag creation" would be great. Such that if you "create" the release in draft in this repo (by adding the tag), then the build is kicked off, and the artifacts are then uploaded to that release (matched by tag).
> 
> Then the release can be published.

Sorry for my late response, I forgot about this...

Is just the opposite. When you want to create a release (and tag the code, of course), you don't need to create the draft. You start this action, and it will create the draft, the tag, build the artifacts and let you complete the draft before publishing it.

I don't know if what you want is to tag the code, and then this start automatically, but I see it easier like it is now. But if you prefer the opposite, I can look at it, but I don't know how difficult will be.

**Review by blckmn — APPROVED** [review]

**Pull request merged**
